crates/bevy_render/src/pipeline/pipeline_compiler.rs

```diff
@@ -282,4 +282,22 @@ impl PipelineCompiler {
             })
             .flatten()
     }
+
+    /// Remove any specialized shaders or pipelines that match the
+    /// changed shader set.
```

Comment:
I would remove "changed" here. In this local scope we don't know _why_ the shaders in the set should be removed, just that they should.